pkg/cloudprovider/aws/instance.go

```diff
@@ -254,35 +254,31 @@ func (p *InstanceProvider) getInstance(ctx context.Context, id string) (*ec2.Ins
 	return instance, nil
 }
 
-func (p *InstanceProvider) instanceToNode(ctx context.Context, instance *ec2.Instance, instanceTypes []cloudprovider.InstanceType) *v1.Node {
+func (p *InstanceProvider) instanceToNode(ctx context.Context, instance *ec2.Instance, instanceTypes []cloudprovider.InstanceType, amiFamily *string) *v1.Node {
 	for _, instanceType := range instanceTypes {
 		if instanceType.Name() == aws.StringValue(instance.InstanceType) {
 			nodeName := strings.ToLower(aws.StringValue(instance.PrivateDnsName))
 			if injection.GetOptions(ctx).GetAWSNodeNameConvention() == options.ResourceName {
 				nodeName = aws.StringValue(instance.InstanceId)
 			}
 
+			// Since we no longer pre-bind, we need to ensure that all resources that were possibly considered for scheduling
+			// purposes are placed on the node.  The extended resources will be moved to an annotation, while the
+			// non-extended will be left on the node.
```

Comment:
It's generally easier to reason about things if we can avoid bimodal behavior (e.g. ABC except for XYZ). Why can't we write these to the node capacity? Is it because the device driver may briefly zero it out and cause overscaling? Alternatively can we store them all in the annotation?

Reply:
We could also not store them and just retrieve them from the instance type

Reply:
Not following, this change just ensures that all resources reported by the instance are also reported on the node object it returns. It was a bug in our AWS provider that we didn't report all resources on the node.  We later peel off the extended resources and store them in an annotation.  We're not guaranteed to see the same instances returned by the instance provider, if an instance type becomes unlaunchable it will disappear.  We've then lost knowledge of what extended resource it had, so I thought it was cleaner just to store the resources on the node as an annotation.

Reply:
Yes, kubelet zeros out both the capacity and the allocatable for extended resources on startup.